File: src/main/java/gargoyle/ct/ver/impl/CTVersionInfoImpl.java
package gargoyle.ct.ver.impl;

import gargoyle.ct.messages.impl.CTMessages;
import gargoyle.ct.ver.CTVersionInfo;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class CTVersionInfoImpl implements CTVersionInfo {
    private static final String KEY_ORGANIZATION_NAME = "organization_name";
    private static final String KEY_ORGANIZATION_URL = "organization_url";
    private static final String KEY_PROJECT_DESCRIPTION = "project_description";
    private static final String KEY_PROJECT_NAME = "project_name";
    private static final String KEY_PROJECT_VERSION = "project_version";
    private static final String LOC_VERSION = "messages/version";
    @NotNull
    private final CTMessages version;

    public CTVersionInfoImpl() {
        version = new CTMessages(LOC_VERSION);
    }

    @Nullable
    @Override
    public String getOrganizationName() {
        return version.getMessage(KEY_ORGANIZATION_NAME);
    }

    @Nullable
    @Override
    public String getOrganizationUrl() {
        return version.getMessage(KEY_ORGANIZATION_URL);
    }

    @Nullable
    @Override
    public String getProjectDescription() {
        return version.getMessage(KEY_PROJECT_DESCRIPTION);
    }

    @Nullable
    @Override
    public String getProjectName() {
        return version.getMessage(KEY_PROJECT_NAME);
    }

    @Nullable
    @Override
    public String getProjectVersion() {
        return version.getMessage(KEY_PROJECT_VERSION);
    }
}

File: src/main/kotlin/gargoyle/ct/ver/impl/CTVersionInfoImpl.kt
package gargoyle.ct.ver.impl

import gargoyle.ct.messages.impl.CTMessages
import gargoyle.ct.ver.CTVersionInfo

class CTVersionInfoImpl : CTVersionInfo {
    private val version: CTMessages = CTMessages(LOC_VERSION)

    override val organizationName: String
        get() = version.getMessage(KEY_ORGANIZATION_NAME)
    override val organizationUrl: String
        get() = version.getMessage(KEY_ORGANIZATION_URL)
    override val projectDescription: String
        get() = version.getMessage(KEY_PROJECT_DESCRIPTION)
    override val projectName: String
        get() = version.getMessage(KEY_PROJECT_NAME)
    override val projectVersion: String
        get() = version.getMessage(KEY_PROJECT_VERSION)

    companion object {
        private const val KEY_ORGANIZATION_NAME = "organization_name"
        private const val KEY_ORGANIZATION_URL = "organization_url"
        private const val KEY_PROJECT_DESCRIPTION = "project_description"
        private const val KEY_PROJECT_NAME = "project_name"
        private const val KEY_PROJECT_VERSION = "project_version"
        private const val LOC_VERSION = "messages/version"
    }
}
